tests/tacacs/test_authorization.py

```diff
@@ -0,0 +1,342 @@
+    ssh = paramiko.SSHClient()
+    ssh.set_missing_host_key_policy(paramiko.AutoAddPolicy())
+    ssh.connect(remote_ip, username=remote_username, password=remote_password, allow_agent=False, look_for_keys=False, auth_timeout=TIMEOUT_LIMIT)
+    return ssh
+    
+
+def check_ssh_connect_remote_failed(remote_ip, remote_username, remote_password):
+    login_failed = False
+    try:
+        ssh_client_local = ssh_connect_remote(remote_ip, remote_username, remote_password)
+    except paramiko.ssh_exception.AuthenticationException as e:
+        login_failed = True
+    
+    pytest_assert(login_failed == True)
+
+def ssh_run_command(ssh_client, command):
+    stdin, stdout, stderr = ssh_client.exec_command(command, timeout=TIMEOUT_LIMIT)
+    stdout_lines = stdout.readlines()
+    stderr_lines = stderr.readlines()
+    return stdout_lines, stderr_lines
+
+def check_ssh_output(res, exp_val):
+    content_exist = False
+    for l in res:
+        if exp_val in l:
+            content_exist = True
+            break
+    pytest_assert(content_exist)
+
+def test_authorization_tacacs_only(localhost, duthosts, enum_rand_one_per_hwsku_hostname, creds_all_duts, check_tacacs):
+    duthost = duthosts[enum_rand_one_per_hwsku_hostname]
+    duthost.shell("sudo config aaa authorization tacacs+")
+    dutip = duthost.host.options['inventory_manager'].get_host(duthost.hostname).vars['ansible_host']
+
+    ssh_client = ssh_connect_remote(dutip, creds_all_duts[duthost]['tacacs_authorization_user'],
+                         creds_all_duts[duthost]['tacacs_authorization_user_passwd'])
+
+    """
+        Verify TACACS+ user run command in server side whitelist:
+            If command have local permission, user can run command.
+    """
+    stdout, stderr = ssh_run_command(ssh_client, "show aaa")
+    check_ssh_output(stdout, 'AAA authentication')
+
+    """
+        Verify TACACS+ user run command in server side whitelist:
+            If command not have local permission, user can't run command.
+    """
+    stdout, stderr = ssh_run_command(ssh_client, "config aaa")
+    check_ssh_output(stderr, 'Root privileges are required for this operation')
+
+    # Verify TACACS+ user can't run command not in server side whitelist.
+    stdout, stderr = ssh_run_command(ssh_client, "cat /etc/passwd")
+    check_ssh_output(stdout, '/usr/bin/cat authorize failed by TACACS+ with given arguments, not executing')
+
+    # Verify Local user can't login.
+    check_ssh_connect_remote_failed(dutip, creds_all_duts[duthost]['tacacs_local_user'],
+                             creds_all_duts[duthost]['tacacs_local_user_passwd'])
+
+    ssh_client.close()
+
+def test_authorization_tacacs_only_some_server_down(localhost, duthosts, enum_rand_one_per_hwsku_hostname, creds_all_duts,ptfhost, check_tacacs):
+    """
+        Setup multiple tacacs server for this UT.
+        Tacacs server 1.2.3.4 not accessible.
+    """
+    invalied_tacacs_server_ip = "1.2.3.4"
```

Comment:
>"1.2.3.4" [](http://example.com/codeflow?start=32&length=9)

How do you know it invalid?

Reply:
change to 127.0.0.1
Because the SONiC will not install tacacs server.